src/schema.rs

```diff
@@ -499,6 +509,112 @@ impl Serialize for RecordField {
+fn parsing_canonical_form(schema: &serde_json::Value) -> String {
+    match schema {
+        serde_json::Value::Number(n) => pcf_float(n),
+        serde_json::Value::Object(map) => pcf_map(map),
+        serde_json::Value::String(s) => pcf_string(s),
+        serde_json::Value::Array(v) => pcf_array(v),
+        _ => unreachable!(),
+    }
+}
+
+fn pcf_map(schema: &Map<String, serde_json::Value>) -> String {
+    use std::borrow::Cow;
+    // Look for the namespace variant up front.
+    let ns = schema.get("namespace").and_then(|v| v.as_str());
+    let mut fields = Vec::new();
+    for (k, v) in schema {
+        // Reduce primitive types to their simple form. ([PRIMITIVE] rule)
+        if schema.len() == 1 && k == "type" {
+            // Invariant: function is only callable from a valid schema, so this is acceptable.
+            if let serde_json::Value::String(s) = v {
+                return pcf_string(s);
+            }
+        }
+
+        // Strip out unused fields ([STRIP] rule)
+        if field_ordering_position(k).is_none() {
+            continue;
+        }
+
+        // Fully qualify the name, if it isn't already ([FULLNAMES] rule).
+        if k == "name" {
+            // Invariant: Only valid schemas. Must be a string.
+            let name = v.as_str().unwrap();
+            let n = match ns {
+                Some(namespace) if !name.contains('.') => {
+                    Cow::Owned(format!("{}.{}", namespace, name))
+                }
+                _ => Cow::Borrowed(name),
+            };
+
+            fields.push((k, format!("{}:{}", pcf_string(k), pcf_string(&*n))));
+            continue;
+        }
+
+        // Strip off quotes surrounding "size" type, if they exist ([INTEGERS] rule).
+        if k == "size" {
+            let i = match v.as_str() {
+                Some(s) => s.parse::<i64>().expect("Only valid schemas are accepted!"),
+                None => v.as_i64().unwrap(),
+            };
+            fields.push((k, format!("{}:{}", pcf_string(k), i)));
+            continue;
+        }
+
+        // For anything else, recursively process the result.
+        fields.push((
+            k,
+            format!("{}:{}", pcf_string(k), parsing_canonical_form(v)),
+        ));
+    }
+
+    // Sort the fields by their canonical ordering ([ORDER] rule).
+    fields.sort_unstable_by_key(|(k, _)| field_ordering_position(k).unwrap());
+    let inter = fields
+        .into_iter()
+        .map(|(_, v)| v)
+        .collect::<Vec<_>>()
+        .join(",");
+    format!("{{{}}}", inter)
+}
+
+fn pcf_array(arr: &[serde_json::Value]) -> String {
+    let inter = arr
+        .iter()
+        .map(parsing_canonical_form)
+        .collect::<Vec<String>>()
+        .join(",");
+    format!("[{}]", inter)
+}
+
+fn pcf_string(s: &str) -> String {
+    format!("\"{}\"", s)
+}
+
+fn pcf_float(schema: &serde_json::Number) -> String {
```

Comment:
The name of this function is a bit misleading as it's actually considering the `Number` as a `i64`.

Also, I don't think any valid schema can only be a JSON number. The only valid number seems to be in the `size` value of a `fixed`, but you're already dealing with it in L559.

Do you think we could get rid of this function and remove the `Number` in `parsing_canonical_form`?

Reply:
You are correct. Removed `pcf_float` and the `Number` match arm. Thanks